### DISCUSSION TITLE
52. 호스트에 IP 주소를 할당하는 방법으로 무엇이 있나요?

### DISCUSSION THREAD

**Discussion by linirini** [jewelry-box/cs-interview#54] [General]
`키워드`: IP 주소, IP 주소 할당, 정적 할당, 동적 할당, DHCP

**Comment by cutehumanS2** [comment]
호스트에 IP 주소를 할당하는 방법에는 크게 **‘정적 할당, ‘동적 할당**’, 두 가지 방법이 있습니다.

- **정적 할당**은 호스트에 직접 수작업으로 IP 주소를 부여하는 방식입니다. 주로 서버나 네트워크 장비처럼 고정된 IP가 필요한 장치에서 사용됩니다.
- **동적 할당**은 일반적으로 DHCP를 통해 자동으로 IP 주소를 할당합니다. 관리가 편하고 IP 주소를 효율적으로 사용할 수 있다는 점에서 일반 사용자 장치에서 많이 사용됩니다.

**Comment by cutehumanS2** [comment]
### 추가 질문
**[오늘날 동적 IP 할당이 주로 사용되는 이유는 무엇인가요?]**

오늘날 동적 IP 할당이 주로 사용되는 이유는, IP 주소 관리의 편리함과 효율성때문입니다.

- DHCP를 통해 자동으로 IP 주소를 할당하기 때문에, 일일이 수작업으로 IP 주소를 설정하지 않아도 됩니다. 대규모 네트워크여도 새로운 장치가 추가되거나 기존 장치를 제거할 때 IP 주소를 편리하게 관리할 수 있습니다.
- 또한, IP 주소 충돌을 방지합니다. 정적 할당은 IP 주소를 중복으로 설정할 가능성이 있지만, DHCP는 이를 자동으로 관리해 충돌을 최소화합니다.
- 스마트폰이나 노트북 같은 이동 장치는 동적 할당이 더 적합합니다. 네트워크에 연결될 때마다 새로운 IP 주소가 필요하기 때문입니다.

**[DHCP 관련 질문 더 보기]**
[3. DHCP에 대해 설명해 주세요.](https://github.com/jewelry-box/cs-interview/discussions/9)

<br><br>

[참고한 자료]
<혼자 공부하는 네트워크>  03-2. IP 주소 中 정적 IP 주소와 동적 IP 주소
